derive-tag/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn;
+
+#[proc_macro_derive(ToTag)]
+pub fn to_tag_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).unwrap();
+    impl_to_tag(&ast)
+}
+
+fn impl_to_tag(ast: &syn::DeriveInput) -> TokenStream {
+    let name = &ast.ident;
+
+    let gen = quote! {
+        impl ToTag for #name {
```

Comment:
It's a good practice in proc macros to always use fully qualified names so people don't redefine them, like this: https://github.com/dtolnay/proc-macro-workshop/blob/master/builder/tests/09-redefined-prelude-types.rs

So **I think** here it would be `to_tag::ToTag`.

Reply:
Maybe doing `std::u8` is overkill though, I would just do it for the `ToTag` 😉